sdks/java/core/src/main/java/org/apache/beam/sdk/io/AvroIO.java

```diff
@@ -962,34 +856,32 @@ private AvroIO() {}
     private final SerializableAvroCodecFactory codec;
     private final ImmutableMap<String, Object> metadata;
 
-    @VisibleForTesting
-    AvroSink(
-        FilenamePolicy filenamePolicy,
-        AvroCoder<T> coder,
-        SerializableAvroCodecFactory codec,
-        ImmutableMap<String, Object> metadata) {
-      super(filenamePolicy);
-      this.coder = coder;
-      this.codec = codec;
-      this.metadata = metadata;
+    private static class ExtractDirectory implements SerializableFunction<ResourceId, ResourceId> {
+      @Override
+      public ResourceId apply(ResourceId input) {
+        return input.getCurrentDirectory();
+      }
```

Comment:
Maybe belongs somewhere more common? Seems like it might be useful elsewhere (e.g. TextIO)